### ISSUE TITLE
ImplementInterfaceRefactoring: Public variables should also be implemented

### ISSUE THREAD

**Issue by ghost** [rubberduck-vba/Rubberduck#1566]
The current implementation of `ImplementInterfaceRefactoring` doesn't implement public variables. Here's the relevant section in [5.2.4.2 Implements Directive](https://msdn.microsoft.com/en-us/library/ee177434.aspx):

```
* A class module containing an <implements-directive> must contain an implemented name declaration corresponding to each public variable declaration contained within the interface class’ class module. The set of required implemented name declarations depends upon of the declared type of the public variable as follows:
* If the declared type of the variable is Variant there must be three corresponding implemented name declarations including a <property-get-declaration>, a <property-let-declaration>, and a <property-set-declaration>.
* If the declared type of the variable is Object or a named class there must be two corresponding implemented name declarations including a <property-get-declaration> and a <property-set-declaration>.
* If the declared type of the variable is anything else, there must be two corresponding implemented name declarations including a <property-get-declaration> and a <property-let-declaration>.
```

**Comment by retailcoder** [comment]
Do I read it correctly?

    'ISomething
    Public Foo As Integer



    'Class1
    Implements ISomething

    Private Property Get ISomething_Foo() As Integer
    End Property

    Private Property Let ISomething_Foo(ByVal value As Integer)
    End Property

That seems rather counter-intuitive, I'd think fields are not allowed on interfaces, and if you want to expose a property you expose, well, a property... have I been doing it wrong all this time?

**Comment by comintern** [comment]
I would say that everyone who uses that syntax has been doing it wrong.  This might warrant something like an InterfaceExposesPublicVariable inspection.

**Comment by autoboosh** [comment]
Yes, I think that's correct. If the variable is of type Variant you'd have to add a Property Set as well. I don't think the properties have to be private either.

**Comment by retailcoder** [comment]
A public field in a class module would already trip the "encapsulate public field" inspection, and since public fields must be implemented as properties then the quickfix for that inspection won't (shouldn't) break anything; I don't think it hurts to change *extract interface* to allow public fields, other than looking very awkward to anyone with a .net background.

**Comment by Hosch250** [comment]
So, should this be changed, or no?

**Comment by retailcoder** [comment]
Not sure when, but this was implemented.
